src/main/java/org/reaktivity/nukleus/maven/plugin/internal/generate/ListFlyweightGenerator.java

```diff
@@ -493,33 +555,104 @@ public Builder build()
         }
     }
 
+    private static final class LengthMethodGenerator extends ClassSpecMixinGenerator
+    {
+        private LengthMethodGenerator(
+            ClassName thisType,
+            Builder builder,
+            AstType superType,
+            TypeResolver resolver)
+        {
+            super(thisType, builder);
+            if (superType != null)
+            {
+                builder.addMethod(methodBuilder("length")
+                    .addModifiers(PUBLIC)
+                    .returns(int.class)
+                    .addAnnotation(Override.class)
+                    .addStatement("return $L.length()", variantRO(resolver.resolveClass(superType)))
+                    .build());
+            }
+        }
+    }
+
+    private static final class FieldCountMethodGenerator extends ClassSpecMixinGenerator
+    {
+        private FieldCountMethodGenerator(
+            ClassName thisType,
+            Builder builder,
+            AstType superType,
+            TypeName logicalLengthType,
+            TypeResolver resolver)
+        {
+            super(thisType, builder);
+            if (superType == null)
+            {
+                builder.addMethod(methodBuilder("fieldCount")
+                    .addModifiers(PUBLIC)
+                    .returns(int.class)
+                    .addStatement("return buffer().$L(offset() + $L)", GETTER_NAMES.get(logicalLengthType),
+                        offset(LOGICAL_LENGTH))
+                    .build());
+            }
+            else
+            {
+                builder.addMethod(methodBuilder("fieldCount")
+                    .addModifiers(PUBLIC)
+                    .returns(int.class)
+                    .addAnnotation(Override.class)
+                    .addStatement("return $L.fieldCount()", variantRO(resolver.resolveClass(superType)))
+                    .build());
+            }
+        }
+    }
+
+    private static final class FieldsMethodGenerator extends ClassSpecMixinGenerator
+    {
+        private FieldsMethodGenerator(
+            ClassName thisType,
+            Builder builder,
+            AstType superType,
+            TypeResolver resolver)
+        {
+            super(thisType, builder);
+            if (superType != null)
+            {
+                builder.addMethod(methodBuilder("fields")
+                    .addModifiers(PUBLIC)
+                    .returns(DIRECT_BUFFER_TYPE)
+                    .addAnnotation(Override.class)
+                    .addStatement("return $L.fields()", variantRO(resolver.resolveClass(superType)))
+                    .build());
+            }
```

Comment:
In general, shouldn't logic to modify builder be deferred until the call to `mixin(builder)` ?
This feedback applies to all `ClassSpecMixinGenerator` subclasses.

Reply:
May I ask the reason behind why it's better to be deferred until the call to `mixin(builder)`? Just wondering.

Reply:
Sure thing.

Most of the generation code uses `ClassSpecGenerator`, and `MethodSpecGenerator` to unconditionally generate complete `class` definitions with specific `method` definitions.

`ClassSpecMixinGenerator` is also used to provide an abstraction that augments the `ClassSpecGenerator` when the `class` definition itself needs to be updated in various ways, such as adding constants or fields in addition to a method, or if the method itself is only conditionally added.

The intent is that the constructor should not have side effects and that the work of mutating the underlying `TypeSpec.Builder` should instead be deferred to the `build()` method, which is even better than adding a separate `mixin(...)` method.

The general pattern is to construct each generator without side effects, then provide details of each member in the case of a struct for example, then build at the end.